### PR TITLE
Add [npm] type definitions badge

### LINKED ISSUE #1252: Badge for modules that come with TypeScript type definitions
Would be nice to have a badge to indicate that a module comes with TypeScript type definitions. This can either be a package written in TypeScript or one written in plain JavaScript with a handwritten type definition file.

I'm thinking something along the line of:

- `[TypeScript friendly]`
- `[TypeScript ready]`

For example, [this module](https://github.com/chalk/chalk/blob/master/types/index.d.ts) comes with handwritten type definitions and could have used such badge.

### PR DESCRIPTION
**Pull request by paulmelnikow**
Let a package show off its type definitions based on devDependency data in the published npm package.

Close #1252

### CODE REVIEW

**Comment by paulmelnikow** [comment]
https://shields-staging-pr-1541.herokuapp.com/npm/types/chalk.svg


https://shields-staging-pr-1541.herokuapp.com/npm/types/vue.svg


(Please ignore the font; that's due to #1309)

File: server.js [hunk 1]
```diff
@@ -1897,6 +1883,55 @@ cache({
...
+    // e.g. cycle, core, svg
+    const [, scope, packageName, format ] = match;
+    const apiUrl = makeNpmPackageDataUrl({
+      registryUrl: queryParams.registry_uri,
+      scope,
+      packageName,
+    });
+    const badgeData = getBadgeData('type definitions', queryParams);
+    request(apiUrl, { headers: { 'Accept': '*/*' } }, function(err, res, buffer) {
+      if (err != null) {
```
Comment (RedSparr0w) [inline server.js]: Could we update this to use the [`checkErrorResponse`](https://github.com/badges/shields/blob/3cb71037fdbed20f74a18880ed9edad4273a08c5/lib/error-helper.js#L3-L19) helper:
https://github.com/badges/shields/blob/75503ec5144b349ab42c4877b774d3acac8157f3/server.js#L6515-L6518

**Comment by paulmelnikow** [comment]
Thanks for the review!

**Comment by sindresorhus** [comment]
I don't think the meaning of the TypeScript version is clear enough. I initially thought it was the latest supported TypeScript version, but it's rather the minimum supported version.

**Comment by paulmelnikow** [comment]
Hmm... I would have thought the version in devDependencies would be the latest required.

Is there a way to infer the latest?

**Comment by paulmelnikow** [comment]
@sindresorhus I'm not familiar enough with TypeScript to know how to identify the supported version range, or to have a project signal which versions they support. Would you have a suggestion?

I would love to get this shipped!

**Comment by paulmelnikow** [comment]
Going to land this. I'd be happy to improve on it in the future!

**Comment by axelpale** [comment]
Link to the badge documentation: https://shields.io/badges/npm-type-definitions

**Pull request merged**